rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,206 @@
+            msg_aggregator: 'MessagesAggregator',
+            router_address: ChecksumEvmAddress,
+            fee_receiver_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(evm_inquirer, base_tools, msg_aggregator)
+        self.router_address = router_address
+        self.fee_receiver_address = fee_receiver_address
+
+    def _decode_swap(
+            self,
+            context: DecoderContext,
+            receiver: ChecksumEvmAddress,
+            sender: ChecksumEvmAddress,
+    ) -> DecodingOutput:
+        """This function is used to decode the swap done by paraswap.
+        At the moment it does not decode the fee in ETH. For example:
+        etherscan.io/tx/0xeeea20b39f157fe59fa4904fd4b62f8971188b53d05c6831e0ed67ee157e40c2
+        TODO Note: https://github.com/orgs/rotki/projects/11?pane=issue&itemId=49582891"""
+        if not self.base.any_tracked((sender, receiver)):
+            return DEFAULT_DECODING_OUTPUT
+
+        out_event: EvmEvent | None = None
+        in_event: EvmEvent | None = None
+        partial_refund_event: EvmEvent | None = None
+        fee_event: EvmEvent | None = None
+        for event in context.decoded_events:
+            if sender != event.location_label:  # in this case, it's not a valid send/receive event
+                continue
+            if (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.SPEND or
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):  # find the send event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.SPEND
+                event.address = self.router_address
+                # not modifying event.notes here since that's done after partial refund below
+                out_event = event
+            elif (
+                event.event_type == HistoryEventType.TRADE and
+                event.event_subtype == HistoryEventSubType.RECEIVE or
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):  # find the receive event
+                event.counterparty = CPT_PARASWAP
+                event.event_type = HistoryEventType.TRADE
+                event.event_subtype = HistoryEventSubType.RECEIVE
+                event.notes = f'Receive {event.balance.amount} {event.asset.resolve_to_asset_with_symbol().symbol} as the result of a swap in paraswap'  # noqa: E501
+                event.address = self.router_address
+                if in_event is None:
+                    in_event = event
+                else:  # if another in_event is found, then it is a partial refund of in_asset
+                    partial_refund_event = event
+
+        if in_event is None or out_event is None:
```

Comment:
nitpick: Easier to read

```suggestion
        if None in (in_event, out_event):
```